problem9/src/main/java/problem9/CustomLinkedList.java

```diff
@@ -0,0 +1,103 @@
+package problem9;
+
+import java.util.HashSet;
+import java.util.Iterator;
+import java.util.Set;
+
+import problem9.CustomLinkedList;
+
+public class CustomLinkedList <T> {
```

Comment:
OK, I see this CustomLinkedList  is everywhere :) So let's move this linkedlist to separate subproject and include it as dependency. As you know there is famous principle DRY so lets avoid any copy/paste stuf.